extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcProcessor.java

```diff
@@ -759,6 +762,14 @@ void registerContextPropagation(ArcConfig config, BuildProducer<ThreadContextPro
         }
     }
 
+    @BuildStep
+    void registerPreShutdownListener(ShutdownBuildTimeConfig shutdownBuildTimeConfig,
+            BuildProducer<ShutdownListenerBuildItem> shutdownListenerBuildItemBuildProducer) {
+        if (shutdownBuildTimeConfig.delayEnabled()) {
```

Comment:
There is also `shutdownBuildTimeConfig#isDelayEnabled()` which doesn't work in development mode.
I chose `delayEnabled()` as the more permissive option.